src/graphql/data/SearchTokens.ts

```diff
@@ -41,19 +42,30 @@ gql`
   }
 `
 
+const ARB_ADDRESS = ARB.address.toLowerCase()
+
 export type SearchToken = NonNullable<NonNullable<SearchTokensQuery['searchTokens']>[number]>
 
-function isMoreRevelantToken(current: SearchToken, existing: SearchToken | undefined, searchChain: Chain) {
-  if (!existing) return true
+/* Returns the more relevant cross-chain token based on native status and search chain */
+function dedupeCrosschainTokens(current: SearchToken, existing: SearchToken | undefined, searchChain: Chain) {
+  if (!existing) return current
+  invariant(current.project?.id === existing.project?.id, 'Cannot dedupe tokens within different tokenProjects')
+
+  // Special case: always prefer Arbitrum ARB over Mainnet ARB
+  if (current.address?.toLowerCase() === ARB_ADDRESS) return current
+  if (existing.address?.toLowerCase() === ARB_ADDRESS) return existing
```

Comment:
Are you sure both of these are necessary? Having just the first check, `if (current.address?.toLowerCase() === ARB_ADDRESS) return current`, with the lowercase casting works for mainnet and arbitrum networks
![Screen Shot 2023-05-18 at 13 42 25 ](https://github.com/Uniswap/interface/assets/11512321/ed65ac3c-7547-464b-a979-9a01b51373eb)

Reply:
It depends on the order the backend returns the two ARB tokens in, if the backend started returning them in the opposite order, only having one check wouldn't work.